source/ui/AMLinePropertyEditor.cpp

```diff
@@ -20,64 +20,173 @@ along with Acquaman.  If not, see <http://www.gnu.org/licenses/>.
 }
 
 AMLinePropertyEditor::AMLinePropertyEditor(const QPen& initialPen, bool areaFilled, const QBrush& areaFillColor, QWidget* parent) :
-	QFrame(parent), ui_(new Ui::AMLinePropertyEditor)
+	QFrame(parent)
 {
-	ui_->setupUi(this);
-
-	ui_->colorButton->setColor(initialPen.color());
-	ui_->widthSlider->setValue(initialPen.width());
-	ui_->lineStyleBox->setCurrentIndex(qBound(0, initialPen.style()-1, 4));
+	setupUi();
 
-	ui_->fillCheckBox->setChecked(areaFilled);
-	ui_->fillColorButton->setColor(areaFillColor.color());
+	colorButton->setColor(initialPen.color());
+	widthSlider->setValue(initialPen.width());
+	lineStyleBox->setCurrentIndex(qBound(0, initialPen.style()-1, 4));
 
-	ui_->fillColorButton->setEnabled(areaFilled);
-	connect(ui_->fillCheckBox, SIGNAL(clicked(bool)), ui_->fillColorButton, SLOT(setEnabled(bool)));
+	fillCheckBox->setChecked(areaFilled);
+	fillColorButton->setColor(areaFillColor.color());
 
-	connect(ui_->colorButton, SIGNAL(colorChanged(QColor)), this, SLOT(onPenSettingsChanged()));
-	connect(ui_->widthSlider, SIGNAL(valueChanged(int)), this, SLOT(onPenSettingsChanged()));
-	connect(ui_->lineStyleBox, SIGNAL(currentIndexChanged(int)), this, SLOT(onPenSettingsChanged()));
-
-	connect(ui_->fillCheckBox, SIGNAL(clicked(bool)), this, SIGNAL(areaFilledChanged(bool)));
-	connect(ui_->fillColorButton, SIGNAL(colorChanged(QColor)), this, SLOT(onAreaFillColorChanged(QColor)));
-}
+	fillColorButton->setEnabled(areaFilled);
+	connect(fillCheckBox, SIGNAL(clicked(bool)), fillColorButton, SLOT(setEnabled(bool)));
 
+	connect(colorButton, SIGNAL(colorChanged(QColor)), this, SLOT(onPenSettingsChanged()));
+	connect(widthSlider, SIGNAL(valueChanged(int)), this, SLOT(onPenSettingsChanged()));
+	connect(lineStyleBox, SIGNAL(currentIndexChanged(int)), this, SLOT(onPenSettingsChanged()));
 
-AMLinePropertyEditor::~AMLinePropertyEditor() {
-	delete ui_;
+	connect(fillCheckBox, SIGNAL(clicked(bool)), this, SIGNAL(areaFilledChanged(bool)));
+	connect(fillColorButton, SIGNAL(colorChanged(QColor)), this, SLOT(onAreaFillColorChanged(QColor)));
 }
 
 void AMLinePropertyEditor::onPenSettingsChanged()
 {
-	emit linePenChanged(QPen(ui_->colorButton->color(),
-							 ui_->widthSlider->value(),
-							 (Qt::PenStyle)(ui_->lineStyleBox->currentIndex()+1)));
+	emit linePenChanged(QPen(colorButton->color(),
+							 widthSlider->value(),
+							 (Qt::PenStyle)(lineStyleBox->currentIndex()+1)));
 }
 
 void AMLinePropertyEditor::onAreaFillColorChanged(const QColor &color)
 {
 	emit areaFillBrushChanged(QBrush(color));
 }
+
+void AMLinePropertyEditor::setupUi()
+{
+	if (objectName().isEmpty())
+		setObjectName(QString::fromUtf8("AMLinePropertyEditor"));
+	resize(159, 193);
+	setStyleSheet(QString::fromUtf8("#AMLinePropertyEditor {\n"
+				"	background-color: rgba(0, 0, 0, 179);\n"
+				"}"));
+	setFrameShape(QFrame::NoFrame);
+	setFrameShadow(QFrame::Plain);
+	verticalLayout = new QVBoxLayout(this);
```

Comment:
You'll need to make sure that there are no 'this' pointers in layouts or widgets since that seems to cause weird crashes that are incredibly hard to trace.